src/PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp

```diff
@@ -0,0 +1,115 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+
+// IWYU pragma:  no_include <pup.h>
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace gr {
+/*!
+ * \brief Class containing functions related to transforming tensors between
+ * spherical and Cartesian Kerr-Schild coordinates in Kerr spacetime.
+ *
+ * The Cartesian Kerr-Schild coordinates \f$(x, y, z)\f$ are defined in terms of
+ * the spherical coordinates \f$(r, \theta, \phi)\f$ through
+ *
+ * \f{align*}
+ * x &= (r \cos\phi + a \sin\phi)\sin\theta =
+ * \sqrt{r^2 + a^2}\sin\theta\cos\left(\phi - \tan^{-1}(a/ r)\right)\\
+ * y &= (r \sin\phi - a \cos\phi)\sin\theta =
+ * \sqrt{r^2 + a^2}\sin\theta\sin\left(\phi - \tan^{-1}(a/ r)\right)\\
+ * z &= r \cos\theta.
+ * \f}
+ *
+ * Using the identities
+ *
+ * \f{align*}
+ * r^2 &= \frac{x^2 + y^2 + z^2 - a^2}{2} +
+ * \sqrt{\left(\frac{x^2 + y^2 + z^2 - a^2}{2}\right)^2 + a^2 z^2}\\
+ * \sin\theta &= \sqrt{\frac{x^2 + y^2}{r^2 + a^2}},\quad
+ * \cos\theta = \frac{z}{r}\\
+ * \sin\phi &= \frac{x a + y r}{\sqrt{(x^2 + y^2)(r^2 + a^2)}},\quad
+ * \cos\phi = \frac{x r - y a}{\sqrt{(x^2 + y^2)(r^2 + a^2)}},
+ * \f}
+ * the Jacobian is
+ *
+ * \f{align*}
+ * \dfrac{\partial(x, y, z)}{\partial(r, \theta, \phi)} &=
+ * \left(\begin{array}{ccc}
+ * \dfrac{xr - ya}{r^2 + a^2} &
+ * \dfrac{xz}{r}\sqrt{\dfrac{r^2 + a^2}{x^2 + y^2}} & -y\\
+ * \dfrac{xa + yr}{r^2 + a^2} &
+ * \dfrac{yz}{r}\sqrt{\dfrac{r^2 + a^2}{x^2 + y^2}} & x\\
+ * \dfrac{z}{r} & - r\sqrt{\dfrac{x^2 + y^2}{r^2 + a^2}} & 0
+ * \end{array}\right),
+ * \f}
+ *
+ * which can be used to transform tensors between both coordinate systems.
+ */
+class KerrSchildCoords {
+ public:
+  /// The black hole spin magnitude in units of the black hole mass.
+  struct SpinParameter {
```

Comment:
This class isn't creatable from options, so you don't need option structs.

Reply:
You are right. Unless someone has a reason why this should be creatable from options, I'll just remove the struct.